### PR TITLE
OBSDOCS-702: Clarify available endpoint auth methods in user workload…

### PR DESCRIPTION
**Pull request by eromanova97**
Version(s): `enterprise-4.12` and later

Issue: [OBSDOCS-702](https://issues.redhat.com/browse/OBSDOCS-702)

Link to docs preview: 
[Example service endpoint authentication settings](https://75588--ocpdocs-pr.netlify.app/openshift-enterprise/latest/observability/monitoring/managing-metrics.html#example-service-endpoint-authentication-settings_managing-metrics)

QE review:
- [x] QE has approved this change.

**Additional information:**

### CODE REVIEW

File: modules/monitoring-specifying-how-a-service-is-monitored.adoc
Comment (eromanova97) [inline modules/monitoring-specifying-how-a-service-is-monitored.adoc:61]: NOTE: Because `bearerTokenFile` and `bearerTokenSecret` are not deprecated for OCP 4.12 and 4.13, I am planning to remove this first sentence of this admonition while cherry-picking in the 4.12 and 4.13 versions. The second sentence will stay.

Comment (eromanova97) [reply modules/monitoring-specifying-how-a-service-is-monitored.adoc:61]: I might just specify it more by saying:
```
If you use the `bearerTokenFile` object to configure a bearer token for scraping the target, the `ServiceMonitor` resource is rejected. Use an `authorization` object instead.
```

Comment (juzhao) [reply modules/monitoring-specifying-how-a-service-is-monitored.adoc:61]: I don't thinks we need to change now, see from https://issues.redhat.com/browse/OCPBUGS-31847, we only see the warnings under debug loglevel now, and servicemonitor with `bearerTokenFile` is  Deprecated, not rejected
```
$ oc explain servicemonitor.spec.endpoints.bearerTokenFile
GROUP:      monitoring.coreos.com
KIND:       ServiceMonitor
VERSION:    v1

FIELD: bearerTokenFile <string>

DESCRIPTION:
    File to read bearer token for scraping the target. 
     Deprecated: use `authorization` instead.
```

Comment (eromanova97) [reply modules/monitoring-specifying-how-a-service-is-monitored.adoc:61]: Hello @simonpasquier I wanted to check with you before we close this one. WDYT?

**Review by simonpasquier — COMMENTED** [review]
A few points from me:
* I don't think that we should modify the existing doc and add the `authorization` field as it would break the example.
* Additional examples showing how to configure authorization, Basic-Auth and OAuth2 would be helpful (similar to what we have already for [remote-write](https://docs.openshift.com/container-platform/4.15/observability/monitoring/configuring-the-monitoring-stack.html#example-remote-write-authentication-settings_configuring-the-monitoring-stack)) though I understand that it's a bigger task.
* In the scope of user-defined monitoring, the usage of `bearerTokenFile` isn't possible and indeed the ServiceMonitor object will be discarded by the operator.
* The tools won't show bearerToken and bearerTokenFile as deprecated but we shouldn't use them in the documentation.

File: modules/monitoring-example-service-endpoint-authentication-settings.adoc
Comment (juzhao) [inline modules/monitoring-example-service-endpoint-authentication-settings.adoc:104]:
```suggestion
.Sample YAML authentication with OAuth 2.0
```

File: modules/monitoring-example-service-endpoint-authentication-settings.adoc
Comment (juzhao) [inline modules/monitoring-example-service-endpoint-authentication-settings.adoc:49]:
```suggestion
<1> The key of the secret to select from the specified `Secret` object.
```
```
$ oc explain ServiceMonitor.spec.endpoints.authorization.credentials.key
GROUP:      monitoring.coreos.com
KIND:       ServiceMonitor
VERSION:    v1

FIELD: key <string>

DESCRIPTION:
    The key of the secret to select from.  Must be a valid secret key.
```

**Comment by juzhao** [comment]
LGTM, waiting for others to review

File: modules/monitoring-example-service-endpoint-authentication-settings.adoc
Comment (simonpasquier) [inline modules/monitoring-example-service-endpoint-authentication-settings.adoc:54]: do we even need this line?

File: modules/monitoring-example-service-endpoint-authentication-settings.adoc
Comment (simonpasquier) [inline modules/monitoring-example-service-endpoint-authentication-settings.adoc:118]: not needed?

File: modules/monitoring-example-service-endpoint-authentication-settings.adoc
Comment (simonpasquier) [inline modules/monitoring-example-service-endpoint-authentication-settings.adoc:44]: I'd suggest to add a "fake" port value otherwise users might copy/paste the example and get confused why it doesn't work.
Same remark for all the examples btw.
```suggestion
        name: example-bearer-auth #<2>
    port: web
```

File: modules/monitoring-specifying-how-a-service-is-monitored.adoc
Comment (simonpasquier) [inline modules/monitoring-specifying-how-a-service-is-monitored.adoc:51]: The ServiceMonitor should be defined in the same namespace as where the app/service runs.

File: modules/monitoring-specifying-how-a-service-is-monitored.adoc
Comment (simonpasquier) [inline modules/monitoring-specifying-how-a-service-is-monitored.adoc:52]: The 2 key points are:
* the selector which matches 1 or many services in the same namespace based on their metadata labels. In almost all cases, you want a selector to match only 1 service. 
* the endpoint's port which tells Prometheus which port to scrape (a service may expose multiple ports).

File: modules/monitoring-example-service-endpoint-authentication-settings.adoc
Comment (eromanova97) [reply modules/monitoring-example-service-endpoint-authentication-settings.adoc:118]: Hello @simonpasquier what value exactly is not needed here? Is it the token one? Is it because we specify the tokenURL in the Servicemonitor object?

Also, I used the [remote-write](https://docs.openshift.com/container-platform/4.15/observability/monitoring/configuring-the-monitoring-stack.html#example-remote-write-authentication-settings_configuring-the-monitoring-stack) section as an exmaple, is there an error too then? I
Thank you for clarification!

Comment (simonpasquier) [reply modules/monitoring-example-service-endpoint-authentication-settings.adoc:118]: yes the `token` field should be removed here, and also from the OAuth2 remote-write example.

Comment (eromanova97) [reply modules/monitoring-example-service-endpoint-authentication-settings.adoc:118]: okay, thank you for clarification!

File: modules/monitoring-example-service-endpoint-authentication-settings.adoc
Comment (eromanova97) [inline modules/monitoring-example-service-endpoint-authentication-settings.adoc:50]:
```suggestion
<1> The key that contains the authentication token in the  `Secret` object.
```

**Review by simonpasquier — COMMENTED** [review]
/lgtm

File: modules/monitoring-example-service-endpoint-authentication-settings.adoc
Comment (mburke5678) [inline modules/monitoring-example-service-endpoint-authentication-settings.adoc:13]: I think each of these examples should be separate. The rendered preview with the text inside the `====` looks odd. 

```
The following sample shows....
.Sample blah
[source,yaml]
----
apiVersion: v1
```

File: modules/monitoring-specifying-how-a-service-is-monitored.adoc
Comment (mburke5678) [inline modules/monitoring-specifying-how-a-service-is-monitored.adoc:53]:
```suggestion
<3> Configure a selector to match your service based on its metadata labels.
```

File: modules/monitoring-example-service-endpoint-authentication-settings.adoc
Comment (eromanova97) [reply modules/monitoring-example-service-endpoint-authentication-settings.adoc:13]: Hello @mburke5678 I have tried many ways to make this as easy to read as possible. I find that this method is actually the best, let me show the other cases:

1. Without a heading, it is just many blacks of texts which makes it really hard to navigate. It would be 6 different blocks of text, which is, in my opinion, not the best solution.
image


2. I tried to create headings for each example. This solution works much better, however, I still think the solution with blocks helps it even more to distinguish:
image

3. If I combine the previous two solutions, I will get  the following result. I like this solution the most out of these three.
image


I took into concideration one more factor though, and that is that we already have one solution like this in the Monitoring docs: [Remote write](https://docs.openshift.com/container-platform/4.15/observability/monitoring/configuring-the-monitoring-stack.html#example-remote-write-authentication-settings_configuring-the-monitoring-stack) That is why I chose that solution in the end, to have some consistency in the docs.

WDYT? I still like the current solution the most, but the solution no. 3 could also work, however, it would not be as consistent with our other similar example reference doc.

Comment (mburke5678) [reply modules/monitoring-example-service-endpoint-authentication-settings.adoc:13]: @eromanova97 I like #3 also. I will defer to your opinion what works best for our customers and the docs.

File: modules/monitoring-example-service-endpoint-authentication-settings.adoc
Comment (adellape) [inline modules/monitoring-example-service-endpoint-authentication-settings.adoc:60]: Unique ID:
```suggestion
[id="sample-yaml-basic-auth_{context}"]
== Sample YAML for Basic authentication
```

File: modules/monitoring-example-service-endpoint-authentication-settings.adoc
Comment (adellape) [inline modules/monitoring-example-service-endpoint-authentication-settings.adoc:107]: Unique ID:
```suggestion
[id="sample-yaml-oauth-20_{context}"]
== Sample YAML authentication with OAuth 2.0
```

**Review by adellape — COMMENTED** [review]
Please squash to one commit and let me know when ready. Also a few nits inline if you'd like to get em. Thanks!

**Comment by adellape** [comment]
Dropping merge-review labels since I'll be off review squad next week. Please re-queue for merge when ready. Thank you!

**Review by skopacz1 — APPROVED** [review]
Merge review LGTM!

**Pull request merged**
